app/src/main/java/org/jellyfin/androidtv/ui/preference/dsl/OptionsAction.kt

```diff
@@ -0,0 +1,51 @@
+package org.jellyfin.androidtv.ui.preference.dsl
+
+import android.content.Context
+import androidx.annotation.DrawableRes
+import androidx.annotation.StringRes
+import androidx.preference.Preference
+import androidx.preference.PreferenceCategory
+import java.util.*
+
+/**
+ * Perform a custom action when clicked.
+ */
+class OptionsAction(
+	private val context: Context
+) : OptionsItemMutable<Unit>() {
+	@DrawableRes
+	var icon: Int? = null
+	var content: String? = null
+	var clickListener: Preference.OnPreferenceClickListener? = null
+
+	fun setListener(preferenceClickListener: Preference.OnPreferenceClickListener) {
```

Comment:
Add an type alias for `Preference.OnPreferenceClickListener` so we don't leak androidx.preference classes from the DSL. Not sure if it's possible to add the typealias in the class. Extending the listener might work too if it's not final.

See https://kotlinlang.org/docs/type-aliases.html